File: src/main/java/com/rockstars/service/AddressService.java
package com.rockstars.service;

import java.util.Set;

import com.rockstars.exception.ServiceException;
import com.rockstars.model.Address;

public interface AddressService {
    public Set<Address> list();
    public Address find(Long id);
    public Address create(Address address);
    public Address update(Address address) throws ServiceException;
    public void delete(Long id) throws ServiceException;
    public Address findCustomerAddress(Long addressId);
}

File: frontend/app/scripts/services/address.service.js
'use strict';

/**
 * @ngdoc service
 * @name yoApp.address
 * @description
 * # address
 * Service in the yoApp.
 */
module.exports = function ($http) {

    this.findAll = function () {
        return $http.get('/api/addresses');
    };

    this.find = function (id) {
        return $http.get('/api/addresses/' + id);
    };

    this.create = function (address) {
        return $http.post('/api/addresses', address);
    };

    this.update = function (address) {
        return $http.put('/api/addresses/' + address.id, address);
    };

    this.destroy = function (id) {
        return $http.delete('/api/addresses/' + id);
    };
};
